lidiabrowser/lidia/models.py

```diff
@@ -0,0 +1,89 @@
+from django.db import models
+
+
+class Publication(models.Model):
+    zotero_id = models.CharField(max_length=100, unique=True)
+    attachment_id = models.CharField(max_length=16, unique=True, null=True)
+    title = models.CharField(max_length=255, null=True)
+
+    def __str__(self):
+        return self.title or self.zotero_id
+
+
+class Language(models.Model):
+    code = models.CharField(max_length=3, unique=True)
+    name = models.CharField(max_length=100, null=True)
+
+    def __str__(self):
+        return self.name or self.code
+
+
+class Annotation(models.Model):
+    RELATION_TYPE_CHOICES = [
+        ("", "None"),
+        ("contradicts", "Contradicts"),
+        ("generalizes", "Generalizes"),
+        ("invalidates", "Invalidates"),
+        ("specialcase", "Is a special case of"),
+        ("supports", "Supports"),
+    ]
+
+    zotero_id = models.CharField(max_length=100, unique=True, null=False)
+    parent_attachment = models.ForeignKey(Publication, on_delete=models.CASCADE, to_field='attachment_id', blank=True, null=True)
+    textselection = models.TextField(default='') # Limit how much text we store, or limit displayed text?
+    argname = models.CharField(max_length=100, unique=True, null=True) # Should we enforce unique here or allow and warn?
+    arglang = models.ForeignKey(Language, on_delete=models.SET_NULL, to_field='code', null=True)
+    description = models.TextField(default='')
+    argcont = models.BooleanField(null=True)
+    page_start = models.IntegerField(null=True) # Should be integers, but...
```

Comment:
Good point, there could be cases where there are roman numbers or other things. I think a CharField is fine; I expect these fields are mainly meant to be human-readable?

Reply:
Yes, I expect so.